kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/DeploymentUtils.java

```diff
@@ -280,19 +283,26 @@ public void conditionEvaluated(EvaluatedCondition<PodStatus> condition) {
     }
 
     /**
-     * Gets node port service url.
+     * Gets node port service address.
      *
      * @param namespace the namespace
      * @param serviceName the service name
-     * @return the node port service url
+     * @return the node port service address
      */
-    public static String getNodePortServiceUrl(String namespace, String serviceName) {
-        var nodeIP = kubeClient(namespace).getClient().nodes().list().getItems().get(0).getStatus().getAddresses().get(0).getAddress();
+    public static String getNodePortServiceAddress(String namespace, String serviceName) {
+        var nodes = kubeClient(namespace).getClient().nodes().list().getItems();
```

Comment:
In kube, nodes aren't namespaced, so passing in the namespace is superfluous.

```suggestion
        var nodes = kubeClient().getClient().nodes().list().getItems();
```